### PR TITLE
ddl: fix data race for ddl seq

### LINKED ISSUE #32541: DATA RACE at the globalSeqNum
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
==================
WARNING: DATA RACE
Write at 0x000008bce270 by goroutine 114:
  github.com/pingcap/tidb/ddl.(*ddl).Start.func1()
      /go/tidb/ddl/ddl.go:398 +0xb7
  github.com/pingcap/tidb/kv.RunInNewTxn()
      /go/tidb/kv/txn.go:49 +0x142
  github.com/pingcap/tidb/ddl.(*ddl).Start()
      /go/tidb/ddl/ddl.go:396 +0xb79
  github.com/pingcap/tidb/domain.(*Domain).Init()
      /go/tidb/domain/domain.go:835 +0xef6
  github.com/pingcap/tidb/session.(*domainMap).Get.func1()
      /go/tidb/session/tidb.go:83 +0x694
  github.com/pingcap/tidb/util.RunWithRetry()
      /go/tidb/util/misc.go:65 +0x93
  github.com/pingcap/tidb/session.(*domainMap).Get()
      /go/tidb/session/tidb.go:71 +0x39a
  github.com/pingcap/tidb/session.createSessionWithOpt()
      /go/tidb/session/session.go:2923 +0x75
[2022/02/22 09:35:45.634 +00:00] [INFO] [ddl.go:674] ["[ddl] DDL job is finished"] [jobID=2]
  github.com/pingcap/tidb/session.createSession()
      /go/tidb/session/session.go:2919 +0x64
  github.com/pingcap/tidb/session.runInBootstrapSession()
      /go/tidb/session/session.go:2902 +0x4b
  github.com/pingcap/tidb/session.BootstrapSession()
      /go/tidb/session/session.go:2751 +0x204
  github.com/pingcap/tidb/ddl_test.setupIntegrationSuite()
      /go/tidb/ddl/serial_test.go:87 +0x26f
  github.com/pingcap/tidb/ddl_test.(*testIntegrationSuite).SetUpSuite()
      /go/tidb/ddl/serial_test.go:113 +0x36
  github.com/pingcap/tidb/ddl_test.(*testIntegrationSuite2).SetUpSuite()
      <autogenerated>:1 +0x56
  runtime.call16()
      /usr/local/go/src/runtime/asm_amd64.s:625 +0x48
  reflect.Value.Call()
      /usr/local/go/src/reflect/value.go:339 +0xd7
  github.com/pingcap/check.(*suiteRunner).runFixture.func1()
      /home/prow/go/pkg/mod/github.com/pingcap/check@v0.0.0-20211026125417-57bd13f7b5f0/check.go:799 +0x245
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/prow/go/pkg/mod/github.com/pingcap/check@v0.0.0-20211026125417-57bd13f7b5f0/check.go:739 +0x12b
Previous write at 0x000008bce270 by goroutine 411:
  github.com/pingcap/tidb/ddl.(*worker).writeDDLSeqNum()
      /go/tidb/ddl/ddl_worker.go:455 +0x5dc
  github.com/pingcap/tidb/ddl.(*worker).finishDDLJob()
      /go/tidb/ddl/ddl_worker.go:447 +0x580
  github.com/pingcap/tidb/ddl.(*worker).handleDDLJobQueue.func1()
      /go/tidb/ddl/ddl_worker.go:562 +0x3dd
  github.com/pingcap/tidb/kv.RunInNewTxn()
      /go/tidb/kv/txn.go:49 +0x142
  github.com/pingcap/tidb/ddl.(*worker).handleDDLJobQueue()
      /go/tidb/ddl/ddl_worker.go:528 +0x1db
  github.com/pingcap/tidb/ddl.(*worker).start()
      /go/tidb/ddl/ddl_worker.go:203 +0x6e6
  github.com/pingcap/tidb/ddl.(*ddl).Start·dwrap·13()
      /go/tidb/ddl/ddl.go:387 +0x47
Goroutine 114 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/prow/go/pkg/mod/github.com/pingcap/check@v0.0.0-20211026125417-57bd13f7b5f0/check.go:734 +0x5e4
  github.com/pingcap/check.(*suiteRunner).runFunc()
      /home/prow/go/pkg/mod/github.com/pingcap/check@v0.0.0-20211026125417-57bd13f7b5f0/check.go:751 +0x67
  github.com/pingcap/check.(*suiteRunner).runFixture()
      /home/prow/go/pkg/mod/github.com/pingcap/check@v0.0.0-20211026125417-57bd13f7b5f0/check.go:795 +0x45
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/prow/go/pkg/mod/github.com/pingcap/check@v0.0.0-20211026125417-57bd13f7b5f0/check.go:647 +0x84
Goroutine 411 (running) created at:
  github.com/pingcap/tidb/ddl.(*ddl).Start()
      /go/tidb/ddl/ddl.go:387 +0x9b3
  github.com/pingcap/tidb/domain.(*Domain).Init()
      /go/tidb/domain/domain.go:835 +0xef6
  github.com/pingcap/tidb/session.(*domainMap).Get.func1()
      /go/tidb/session/tidb.go:83 +0x694
  github.com/pingcap/tidb/util.RunWithRetry()
      /go/tidb/util/misc.go:65 +0x93
  github.com/pingcap/tidb/session.(*domainMap).Get()
      /go/tidb/session/tidb.go:71 +0x39a
  github.com/pingcap/tidb/session.createSessionWithOpt()
      /go/tidb/session/session.go:2923 +0x75
  github.com/pingcap/tidb/session.createSession()
      /go/tidb/session/session.go:2919 +0x64
  github.com/pingcap/tidb/session.runInBootstrapSession()
      /go/tidb/session/session.go:2902 +0x4b
  github.com/pingcap/tidb/session.BootstrapSession()
      /go/tidb/session/session.go:2751 +0x204
  github.com/pingcap/tidb/ddl_test.setUpSuite()
      /go/tidb/ddl/db_test.go:121 +0x2fc
  github.com/pingcap/tidb/ddl_test.(*testDBSuite).SetUpSuite()
      /go/tidb/ddl/db_test.go:143 +0x36
  github.com/pingcap/tidb/ddl_test.(*testDBSuite8).SetUpSuite()
      <autogenerated>:1 +0x56
  runtime.call16()
      /usr/local/go/src/runtime/asm_amd64.s:625 +0x48
  reflect.Value.Call()
      /usr/local/go/src/reflect/value.go:339 +0xd7
  github.com/pingcap/check.(*suiteRunner).runFixture.func1()
      /home/prow/go/pkg/mod/github.com/pingcap/check@v0.0.0-20211026125417-57bd13f7b5f0/check.go:799 +0x245
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/prow/go/pkg/mod/github.com/pingcap/check@v0.0.0-20211026125417-57bd13f7b5f0/check.go:739 +0x12b
================== 
```

### PR DESCRIPTION
**Pull request by wjhuang2016**
Signed-off-by: wjhuang2016 <huangwenjun1997@gmail.com>

### What problem does this PR solve?

Issue Number: close #32541

Problem Summary:

### What is changed and how it works?
Move ddlSeqNumMu and ddlSeqNum to ddlCtx
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: ddl/ddl_worker.go [hunk 1]
```diff
@@ -606,9 +605,9 @@ func (w *worker) handleDDLJobQueue(d *ddlCtx) error {
 		if err != nil {
 			if w.lockSeqNum {
 				// txn commit failed, we should reset globalSeqNum
```
Comment (hawkingrei) [inline ddl/ddl_worker.go:607]: This comment need to be updated.

File: ddl/ddl.go [hunk 2]
```diff
@@ -395,7 +400,7 @@ func (d *ddl) Start(ctxPool *pools.ResourcePool) error {
 
 		err = kv.RunInNewTxn(d.ctx, d.store, true, func(ctx context.Context, txn kv.Transaction) error {
 			t := meta.NewMeta(txn)
-			globalSeqNum, err = t.GetHistoryDDLCount()
```
Comment (hawkingrei) [inline ddl/ddl.go:398]: Does it add the ```Lock``` to avoid data race？ it is necessary to add by checking the stack from #32541.

**Review by hawkingrei — APPROVED** [review]

**Review by xiongjiwei — APPROVED** [review]

**Pull request merged**